source/elements/oneDAL/source/data_management/tables.rst

```diff
@@ -1,3 +1,475 @@
+  (see :ref:`Metadata API` section for details).
+
+-------------------------------
+Table types in |dal_short_name|
+-------------------------------
+|dal_short_name| defines a set of classes, each class implements a table contact
+with a specific set of metadata values (see :ref:`Metadata API` for details):
+
+.. list-table::
+  :header-rows: 1
+
+  * - Table type
+    - :ref:`Data layout <Data layout API>`
+    - :ref:`Data format <Data format API>`
+    - Is contiguous
+    - Is homogeneous
+  * - homogen_table_
+    - row_major/column_major
+    - dense
+    - yes
+    - yes
+  * - soa_table_
+    - column_major
+    - dense
+    - yes
+    - yes/no
+  * - aos_table_
+    - row_major
+    - dense
+    - yes
+    - yes/no
+  * - csr_table_
+    - row_major
+    - csr
+    - yes
+    - yes
+
+-----------------------------
+Requirements on table objects
+-----------------------------
+Each table object in |dal_short_name| follow these requirements:
+
+1. Table objects in |dal_short_name| are :term:`immutable <Immutability>` (it is not possible
+   to change data or metadata values inside the table).
+
+2. To create compex table types or modify table data, :ref:`builders <Builders>` should be used.
+
+3. Table objects in |dal_short_name| are reference-counted. One can use an assignment operator or copy constructor
+   on table objects to create another reference to it.
+   ::
+
+      onedal::table table2 = table1;
+      // table1 and table2 share the data (no data copy is performed)
+
+      table3 = table2;
+      // table1, table2 and table3 share the same data
+
+4. Every table type must be inherited from the base ``table`` class, which represents
+   a generalized table API. (see :ref:`Table API` for details)
+
+5. Every table type is implemented over particular set of metadata values and must hide other
+   implementation details from public API.
+
+-------------------------------
+Entities and their dependencies
+-------------------------------
+
+This section describes dependencies between all the classes and structures
+related to tables.
+
+TBD
+
+.. _Table API:
+
+---------
+Table API
+---------
+::
+
+   class table {
+   public:
+      table() = default;
+
+      template <typename TableImpl,
+               typename = std::enable_if_t<is_table_impl_v<TableImpl>>>
+      table(TableImpl&&);
+
+      table(const table&);
+      table(table&&);
+
+      table& operator=(const table&);
+
+      std::int64_t get_feature_count() const noexcept;
+      std::int64_t get_observation_count() const noexcept;
+      bool is_empty() const noexcept;
+      const dal::table_meta& get_metadata() const noexcept;
+   };
+
+.. namespace:: onedal
+.. class:: table
+
+   .. function:: table()
+
+      Creates an empty table with no data and ``table_meta`` constructed by default
+
+   .. function:: table(TableImpl&&)
+
+      Creates a table object using the implementation passes an a parameter
```

Comment:
what do you mean by "passes an a parameter"?

Reply:
a lot of typos, I meant "Creates a table object using the entity passed as a parameter"

Reply:
which entity? is it self-evident?